File: Shortest Path/Bellman-Ford/11657번 - 타임머신/11657.cpp
#include <iostream>
#include <vector>

using namespace std;

#define INF 1e10

int n, m;
vector<vector<int> > edges;
long long min_cost[501];

void input() {
    cin >> n >> m;
    for(int i = 0, a, b, c; i < m; i++) {
        cin >> a >> b >> c;
        edges.push_back({a, b, c});
    }
    
    fill_n(&min_cost[0], 501, INF);
}

void solve() {
    min_cost[1] = 0;
    
    for(int i = 0; i < n; i++) {
        for(auto &edge : edges) {
            int from = edge[0], to = edge[1], cost = edge[2];
            
            if(min_cost[from] == INF || min_cost[to] < min_cost[from] + cost) continue;
            min_cost[to] = min_cost[from] + cost;
        }
    }
    
    for(auto &edge : edges) {
        int from = edge[0], to = edge[1], cost = edge[2];
        
        if(min_cost[from] != INF && min_cost[to] > min_cost[from] + cost) {
            cout << -1 << '\n';
            exit(0);
        }
    }
    
    for(int i = 2; i <= n; i++) cout << (min_cost[i] == INF ? -1 : min_cost[i]) << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Shortest Path/Bellman-Ford/11657번 - 타임머신/11657.py
import sys

input = sys.stdin.readline

INF = 10**9

n, m = map(int, input().split())
edges = [list(map(int, input().split())) for _ in range(m)]
min_cost = [INF] * (n + 1)

def solve():
    min_cost[1] = 0
    
    for _ in range(n):
        for edge in edges:
            st, end, cost = edge[0], edge[1], edge[2]
            nxt_cost = min_cost[st] + cost
            
            if min_cost[st] == INF or min_cost[end] < nxt_cost:
                continue
            min_cost[end] = nxt_cost
            
    for edge in edges:
        st, end, cost = edge[0], edge[1], edge[2]
        nxt_cost = min_cost[st] + cost
        
        if min_cost[st] != INF and min_cost[end] > nxt_cost:
            print(-1)
            exit(0)
                
    for i in range(2, n + 1):
        print(-1 if min_cost[i] == INF else min_cost[i])
                
                

if __name__ == '__main__':
    solve()